### DISCUSSION TITLE
CP using neper mesh

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#26208] [Q&A Modules: Solid mechanics]
Hello, 

I want to use the FiniteStrainCrystalPlasticity material with a neper generated mesh but I am getting a constitutive failure error in the FiniteStrainCrystalPlasticity block. When I use normal Moose generated mesh with Voronoi tessellation I do not get any error and it works fine. Am I missing a step here for using a neper mesh with moose ?

**Comment by GiudGiud** [comment]
@amassaf what's the printout for the error on the console?
@laagesen @dschwen  who uses neper we could tag?

**Reply by ghost** [reply]
*** ERROR ***
The following error occurred in the object "crysp", of type "FiniteStrainCrystalPlasticity".

FiniteStrainCrystalPlasticity: Constitutive failure

**Comment by sapitts** [comment]
The class `FiniteStrainCrystalPlasticity` is no longer actively developed and is deprecated. We recommend using the crystal plasticity capabilities in the `ComputeMultipleCrystalPlasticityStress` and `CrystalPlasticityKalidindiUpdate` combination, which is actively being developed. Please see the documentation [here](https://mooseframework.inl.gov/source/materials/crystal_plasticity/ComputeMultipleCrystalPlasticityStress.html).

**Reply by ghost** [reply]
OK thank you, I will look into it.

**Comment by ngrilli** [comment]
@amassaf  I made a youtube video about this:

https://www.youtube.com/watch?v=qG7EoX34hZ8&t=1s

and there's an example here:

https://github.com/ngrilli/c_pfor_am/tree/main/test/tests/neper

the main issue being reading the node sets, therefore I used the bounding box node set generator

**Reply by ghost** [reply]
Hello, Thank you I will check this out.

**Reply by ghost** [reply]
Hello, thank you so much this has been very useful. I noticed that in your example you use ComputeElasticityTensorCPGrain and for GrainPropertyReadFile to be able to define euler angles, is this still required now after updates to be able to use neper mesh with moose? because the changes in the neper2moose.txt aren't needed anymore, right? Thank you.
